threestudio/utils/config.py

```diff
@@ -20,10 +20,53 @@
 OmegaConf.register_new_resolver("rmspace", lambda s, sub: s.replace(" ", sub))
 OmegaConf.register_new_resolver("tuple2", lambda s: [float(s), float(s)])
 OmegaConf.register_new_resolver("gt0", lambda s: s > 0)
+OmegaConf.register_new_resolver("cmaxgt0", lambda s: C_max(s) > 0)
 OmegaConf.register_new_resolver("not", lambda s: not s)
+OmegaConf.register_new_resolver(
+    "cmaxgt0orcmaxgt0", lambda a, b: C_max(a) > 0 or C_max(b) > 0
+)
 # ======================================================= #
 
 
+def C_max(value: Any) -> float:
+    if isinstance(value, int) or isinstance(value, float):
+        pass
+    else:
+        value = config_to_primitive(value)
+        if not isinstance(value, list):
+            raise TypeError("Scalar specification only supports list, got", type(value))
+        if len(value) == 3:
+            value = [0] + value
+        assert len(value) == 4
+        start_step, start_value, end_value, end_step = value
+        value = max(start_value, end_value)
+    return value
+
+
+def C(value: Any, epoch: int, global_step: int) -> float:
```

Comment:
Why do we need to move `C` here? If we do need to move it, there're many other places where we need to change the import path of `C` accordingly.

Reply:
I imported C within [threestudio/utils/misc.py](https://github.com/threestudio-project/threestudio/pull/193/files/328d46606966004a3de3818c797f351d688fc4b8#diff-385e6ce238ca275f126b9b2bc08a49b68e5391b1f44af1804b7787c83d08aa54) so that should take care of all other imports.

I had imported it initially because I needed it for one of the resolvers. I could remove it now though if it might still cause problems.